### PR TITLE
fix: use flag in preflush to indicate whether reorder is required

### PR DESCRIPTION
**Pull request by jiacai2050**
## Rationale
Current we suggest new primary keys in preflush, and do a reorder in flush task.

There is no easy way for flush task to know if reorder if required.

## Detailed Changes
- Introduce a flag to represent whether reorder is required
- Change `col` to `ident` when build logical plan, `ident` will not normalize column name.

## Test Plan
Integration test.

### CODE REVIEW

File: analytic_engine/src/instance/flush_compaction.rs [hunk 1]
```diff
@@ -196,6 +196,10 @@ pub struct TableFlushRequest {
     pub table_data: TableDataRef,
     /// Max sequence number to flush (inclusive).
     pub max_sequence: SequenceNumber,
+
+    /// We may suggest new primary keys in preflush. if suggest happens, we need
```
Comment (Rachelint) [inline analytic_engine/src/instance/flush_compaction.rs:200]: If suggestion happened

File: analytic_engine/src/sampler.rs [hunk 2]
```diff
@@ -292,7 +292,7 @@ impl PrimaryKeySampler {
             .iter()
             .enumerate()
             .map(|(idx, col)| {
-                if idx == timestamp_index {
+                if col.data_type.is_timestamp() {
```
Comment (Rachelint) [inline analytic_engine/src/sampler.rs:295]: It is possible that, `Timestamp` typed column but not a `time key` is suggested to be one of the order keys?

File: analytic_engine/src/instance/flush_compaction.rs [hunk 3]
```diff
@@ -589,11 +603,13 @@ impl FlushTask {
...
         let timestamp_idx = self.table_data.schema().timestamp_index();
-        if let Some(pk_idx) = self.table_data.current_version().suggest_primary_key() {
+        if need_reorder {
+            let schema = self.table_data.schema();
+            let primary_key_indexes = schema.primary_key_indexes();
             let reorder = Reorder {
                 iter,
                 schema: self.table_data.schema(),
-                order_by_col_indexes: pk_idx,
+                order_by_col_indexes: primary_key_indexes.to_vec(),
```
Comment (Rachelint) [inline analytic_engine/src/instance/flush_compaction.rs:612]: indices

Comment (jiacai2050) [reply analytic_engine/src/instance/flush_compaction.rs:612]: both indexes and indices are valid plural for index, and most of our code base use indexes.
- https://www.merriam-webster.com/dictionary/index

File: analytic_engine/src/sampler.rs (hunk 2 above)
Comment (jiacai2050) [reply analytic_engine/src/sampler.rs:295]: It make no sense for me, I think we should be strict at first.

**Review by Rachelint — APPROVED** [review]
LGTM

**Pull request merged**